server/network/aoprotocol.py

```diff
@@ -48,6 +51,14 @@ def __init__(self, server):
         self.buffer = ''
         self.ping_timeout = None
 
+    def dezalgo(self, input):
+        """
+        Turns any string into a de-zalgo'd version, with a tolerance to allow for special language characters.
+        """
+        print(self.server.zalgo_tolerance)
+        filtered = re.sub("([̀-ͯ᪰-᫿᷀-᷿⃐-⃿︠-︯]{" + re.escape(str(self.server.zalgo_tolerance)) + ",})",'',input)
```

Comment:
Right now, you are representing a lot of code points in a very compressed amount of space. My suggestion to you is to express them as escaped Unicode to avoid confusion when adding more Zalgo characters - e.g. `\u1AB0`. Or you can even refer to the specific glyphs by name (e.g. `\N{GREEK CAPITAL LETTER DELTA}"`).

Reply:
Maybe make a table of banned zalgo characters and match it?

Reply:
I think I implemented exactly what you want in my fork, but I'll need to wait till tonight or tomorrow to test it.